src/javascripts/app.js

```diff
@@ -1,32 +1,35 @@
-(function() {
-  return txApp([
-    require('transifex-api/project'),
-    require('transifex-api/resource'),
-    require('zendesk-api/article'),
-    require('zendesk-api/category'),
-    require('zendesk-api/section'),
-    require('zendesk-api/dynamic-content'),
-    require('zendesk-api/config'),
-    require('zendesk-api/pagination'),
-    require('ui/sync-articles'),
-    require('ui/sync-categories'),
-    require('ui/sync-sections'),
-    require('ui/sync-dynamic-content'),
-    require('ui/brand-project'),
-    require('ui/notifications'),
-  ]);
-}());
-
-var logger = require('logger'),
-    io = require('io'),
-    txutils = require('txUtil');
+import BaseApp from 'base_app';
+
+var App = txApp([
+  require('./transifex-api/project'),
+  require('./transifex-api/resource'),
+  require('./zendesk-api/article'),
+  require('./zendesk-api/category'),
+  require('./zendesk-api/section'),
+  require('./zendesk-api/dynamic-content'),
+  require('./zendesk-api/config'),
+  require('./zendesk-api/pagination'),
+  require('./ui/sync-articles'),
+  require('./ui/sync-categories'),
+  require('./ui/sync-sections'),
+  require('./ui/sync-dynamic-content'),
+  require('./ui/brand-project'),
+  require('./ui/notifications'),
+]);
+
+var logger = require('./logger'),
+    io = require('./io'),
+    txutils = require('./txUtil');
 
 function txApp(modules) {
   // App was activated
   function appActivated() {
     //set settings to be accessible from everywhere
     io.setSettings(this.settings);
-    io.setEmail(this.currentUser().email());
+    this.zafClient.get('currentUser.email').
+      then((data) => {
```

Comment:
I am leaving this comment here as well, just for the record. This call doesn't return before the initialization of `zendesk-api/factory.js` where the email is used, resulting in the wrong authentication header later in the requests.